x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/action/TransportMoveToStepAction.java

```diff
@@ -48,12 +51,50 @@ protected void masterOperation(Task task, Request request, ClusterState state, A
             listener.onFailure(new IllegalArgumentException("index [" + request.getIndex() + "] does not exist"));
             return;
         }
-        clusterService.submitStateUpdateTask("index[" + request.getIndex() + "]-move-to-step",
+
+        if (LifecycleSettings.LIFECYCLE_NAME_SETTING.get(indexMetadata.getSettings()) == null) {
+            listener.onFailure(new IllegalArgumentException("index [" + request.getIndex() + "] is not managed by ILM"));
+            return;
+        }
+
+        final Request.PartialStepKey abstractTargetKey = request.getNextStepKey();
+        final String targetStr = abstractTargetKey.getPhase() + "/" + abstractTargetKey.getAction() + "/" + abstractTargetKey.getName();
+
+        // Resolve the key that could have optional parts into one
+        // that is totally concrete given the existing policy and index
+        Step.StepKey concreteTargetStepKey = indexLifecycleService.resolveStepKey(state, indexMetadata.getIndex(),
+            abstractTargetKey.getPhase(), abstractTargetKey.getAction(), abstractTargetKey.getName());
+
+        // We do a pre-check here before invoking the cluster state update just so we can skip the submission if the request is bad.
+        if (concreteTargetStepKey == null) {
+            // This means we weren't able to find the key they specified
+            listener.onFailure(new IllegalArgumentException("unable to determine concrete " +
+                "step key from target next step key: " + abstractTargetKey));
+            return;
+        }
+
+        clusterService.submitStateUpdateTask("index[" + request.getIndex() + "]-move-to-step-" + targetStr,
```

Comment:
should this use `concreteTargetStepKey` instead of the (potentially) partial `targetStr` ? (to avoid a human having to parse what the actual target is when inspecting the tasks)

Reply:
One issue is that we don't know the *real* concrete target step key until we're in the cluster state update (because there could be changes to the policy between now and when the cluster state update is executed)